src/orange/surf/Involute.hh

```diff
@@ -285,12 +292,13 @@ Involute::calc_intersections(Real3 const& pos,
  */
 CELER_FORCEINLINE_FUNCTION Real3 Involute::calc_normal(Real3 const& pos) const
 {
-    real_type const x = pos[0] - origin_[0];
-    real_type const y = pos[1] - origin_[1];
+    Real2 xy;
+    xy[0] = pos[0] - origin_[0];
+    xy[1] = pos[1] - origin_[1];
 
     // Calculate normal
     real_type const angle
-        = std::sqrt((ipow<2>(x) + ipow<2>(y) / (ipow<2>(r_b_))) - 1) + a_;
+        = std::sqrt((dot_product(xy, xy) / ipow<2>(r_b_)) - 1) + a_;
```

Comment:
There's a chance the point is going to be slightly inside but not exactly on the surface
```suggestion
        = std::sqrt(clamp_to_nonneg(dot_product(xy, xy) / ipow<2>(r_b_) - 1)) + a_;
```